src/cryptography/x509.py

```diff
@@ -1175,6 +1207,71 @@ def public_bytes(self, encoding):
 
 
 @six.add_metaclass(abc.ABCMeta)
+class CertificateRevocationList(object):
+
+    @abc.abstractmethod
+    def fingerprint(self, algorithm):
+        """
+        Returns bytes using digest passed.
+        """
+
+    @abc.abstractproperty
+    def signature_hash_algorithm(self):
+        """
+        Returns a HashAlgorithm corresponding to the type of the digest signed
+        in the certificate.
+        """
+
+    @abc.abstractproperty
+    def issuer(self):
+        """
+        Returns the X509Name with the issuer of this CRL.
+        """
+
+    @abc.abstractproperty
+    def next_update(self):
+        """
+        Returns the date of next update for this CRL.
+        """
+
+    @abc.abstractproperty
+    def last_update(self):
+        """
+        Returns the date of last update for this CRL.
+        """
+
+    @abc.abstractproperty
+    def revoked_certificates(self):
+        """
+        Returns a list of RevokedCertificate objects for this CRL.
+        """
+
+    @abc.abstractproperty
+    def extensions(self):
+        """
+        Returns an Extensions object containing a list of CRL extensions.
+        """
+
+    @abc.abstractmethod
+    def verify(self, pubkey):
+        """
+        Verify this CRL against a given 'pubkey'.
+        """
```

Comment:
Yes, no argument against that here. From the original comment it just
sounded like this sort of verification wasn't considered to be doable at
all.

On Thu, May 28, 2015 at 10:51 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1956#discussion_r31259379>:
>
> > +    def revoked_certificates(self):
> > +        """
> > +        Returns a list of RevokedCertificate objects for this CRL.
> > +        """
> > +
> > +    @abc.abstractproperty
> > +    def extensions(self):
> > +        """
> > +        Returns an Extensions object containing a list of CRL extensions.
> > +        """
> > +
> > +    @abc.abstractmethod
> > +    def verify(self, pubkey):
> > +        """
> > +        Verify this CRL against a given 'pubkey'.
> > +        """
>
> Yeah, I definitely agree that in hazmat the core verification should not
> require network egress. The goal of this project is to build APIs that make
> it hard to shoot yourself in the foot though, so when we add verification
> here if it ends up being the final check (does this public key match the
> signature) we may want to think carefully about what we name that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1956/files#r31259379>.
>